### ISSUE TITLE
Questions for class meeting 9th April

### ISSUE THREAD

**Issue by YPerumal** [iandurbach/exploratory-data-analysis#7]
Hi Ian, I didn't see a thread for Friday's session so hope it's alright I made one.
I have a some questions:
* Megan oftened mentioned becoming a shiny dev in the video. Is this a standalone career in demand? Or more something you have in your toolkit
* How does Shiny compare to the other options out there. There doesn't seem to be a free version in other languages(maybe Dash in python?) which is free, or Tableau(which good but is very not free)
* How exactly is the data loaded and then referenced by the server? it doesn't seem explicitly stated.Or when loading .rda files just put the data frames into memory with those names again?
* In Exercise 5, when rendering the data table in the server, I don't quite understand what the "ungroup" is doing and why we need it?

**Comment by liamcarew** [comment]
R Shiny:

1) I just wanted to clarify if my understanding is correct: The 'ui' function creates an 'input' object with all the reference IDs as attributes and then this object is passed into the server function to render the inputs and create an 'output'  object to send back to the UI to display? And then the 'shiny' function creates the event loop and checks if the input is invalidated or not?

**Comment by iandurbach** [comment]
@YPerumal of course, thanks for starting the issue! Have a participation point +1 :) 

We can chat more in class, but briefly:
- more a tool in your toolkit, I'd say
- can't say much as I haven't used the others. The main benefit to Shiny is giving you a way of turning your R code into a nice-looking web app that you can very easily share with others, so if you're doing most of your analysis in R anyway, Shiny is very attractive. 
- Rda is short for Rdata, so things work as for Rdata files. You can have multiple objects stored in them with specific names, so when you load the file, all those named objects are then available (under those names) to the app.
- the 'group_by' groups observations, unless you ungroup them after the summary (or whatever you've used the grouped data for) you can get some unexpected results!

@liamcarew you're correct!
